### PR TITLE
created new end point for registering a new filing member

### PR DESCRIPTION
**Pull request by alikouhestani**

### CODE REVIEW

File: app/uk/gov/hmrc/pillar2/service/RegistrationService.scala [hunk 1]
```diff
@@ -70,19 +68,32 @@ class RegistrationService @Inject() (
       true
     )
   }.getOrElse {
-    logger.warn(s"[Session ID: ${SessionIdHelper.sessionId(hc)}] - RegistrationService - Filing Member Registration Information Missing")
+    logger.warn("Filing member registration failed as one or more required fields were missing")
     registerWithoutIdError
   }
 
+  def registerNewFilingMember(userAnswers: UserAnswers)(implicit hc: HeaderCarrier) =
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:75]: Add return type annotation.

File: app/uk/gov/hmrc/pillar2/controllers/RegistrationController.scala [hunk 2]
```diff
@@ -54,15 +50,21 @@ class RegistrationController @Inject() (
...
-    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
-    logger.info("[Session ID: ${SessionIdHelper.sessionId(hc)}] - Calling Registration Submission without FmId")
     getUserAnswers(id).flatMap { userAnswer =>
       dataSubmissionService
         .sendNoIdFmRegistration(userAnswer)
         .map(handleResult)
     }
   }
 
+  def registerNewFilingMember(id: String): Action[AnyContent] = authenticate.async { implicit request =>
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/pillar2/controllers/RegistrationController.scala:60]: rename method to registerNoIdNewFilingMember to distinguish between NoId and Id journeys for new filing member.

File: app/uk/gov/hmrc/pillar2/service/RegistrationService.scala (hunk 1 above)
Comment (alikouhestani) [reply app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:75]: Done

Comment (jassalrichy) [reply app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:75]: rename method to registerNoIdNewFilingMember to distinguish between NoId and Id journeys for new filing member.

File: app/uk/gov/hmrc/pillar2/service/RegistrationService.scala [hunk 3]
```diff
@@ -70,19 +68,32 @@ class RegistrationService @Inject() (
...
   private def registerWithoutId(businessName: String, address: Address, contactDetails: ContactDetails, isFm: Boolean)(implicit
-    hc:                                       HeaderCarrier,
-    ec:                                       ExecutionContext
+    hc:                                       HeaderCarrier
   ): Future[HttpResponse] = {
     val registerWithoutIDRequest = RegisterWithoutIDRequest(businessName, address, contactDetails)
     val response = dataSubmissionConnectors
       .sendWithoutIDInformation(registerWithoutIDRequest)(hc, ec)
-    response.map { res =>
-      val resReceived = AuditResponseReceived(res.status, res.json)
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:85]: Just checking, is there no requirement to create an audit response?

File: test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala [hunk 4]
```diff
@@ -218,4 +218,82 @@ class RegistrationControllerSpec extends BaseSpec with Generators with ScalaChec
...
+          )
+        )
+        val request = FakeRequest(POST, rfmRoute(userAnswers.id)).withJsonBody(jsData)
+        val result  = route(application, request).value
+        status(result) mustEqual FORBIDDEN
+
+      }
+    }
+
+    "rerutn INTERNAL_SERVER_ERROR with data has been submitted" in new Setup {
```
Comment (jassalrichy) [inline test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala:282]: correct typo.

File: test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala [hunk 5]
```diff
@@ -218,4 +218,82 @@ class RegistrationControllerSpec extends BaseSpec with Generators with ScalaChec
...
+            HttpResponse.apply(NOT_FOUND, "Not Found")
+          )
+        )
+        val request = FakeRequest(POST, rfmRoute(userAnswers.id)).withJsonBody(jsData)
+        val result  = route(application, request).value
+        status(result) mustEqual NOT_FOUND
+
+      }
+    }
+    "rerutn FORBIDDEN with data has been submitted" in new Setup {
```
Comment (jassalrichy) [inline test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala:267]: correct typo.

File: test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala [hunk 6]
```diff
@@ -218,4 +218,82 @@ class RegistrationControllerSpec extends BaseSpec with Generators with ScalaChec
...
+          )
+        )
+        val request = FakeRequest(POST, rfmRoute(userAnswers.id)).withJsonBody(jsData)
+        val result  = route(application, request).value
+        status(result) mustEqual BAD_REQUEST
+
+      }
+    }
+
+    "rerutn NOT_FOUND with data has been submitted" in new Setup {
```
Comment (jassalrichy) [inline test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala:253]: correct typo.

File: test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala [hunk 7]
```diff
@@ -218,4 +218,82 @@ class RegistrationControllerSpec extends BaseSpec with Generators with ScalaChec
...
+          )
+        )
+        val request = FakeRequest(POST, rfmRoute(userAnswers.id)).withJsonBody(jsData)
+        val result  = route(application, request).value
+        status(result) mustEqual OK
+
+      }
+    }
+
+    "rerutn BAD_REQUEST with data has been submitted" in new Setup {
```
Comment (jassalrichy) [inline test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala:238]: correct typo.

File: test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala [hunk 8]
```diff
@@ -218,4 +218,82 @@ class RegistrationControllerSpec extends BaseSpec with Generators with ScalaChec
 
   }
 
+  "RegisterNewFilingMember" - {
+
+    "rerutn OK with valid data has been submitted" in new Setup {
```
Comment (jassalrichy) [inline test/uk/gov/hmrc/pillar2/controllers/RegistrationControllerSpec.scala:223]: correct typo.

File: app/uk/gov/hmrc/pillar2/service/RegistrationService.scala [hunk 9]
```diff
@@ -70,19 +68,32 @@ class RegistrationService @Inject() (
...
+    response.map { _ =>
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:96]: remove unused import AuditResponseReceived

File: app/uk/gov/hmrc/pillar2/service/RegistrationService.scala [hunk 10]
```diff
@@ -40,7 +40,6 @@ class RegistrationService @Inject() (
 ) extends Logging {
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:40]: remove unused DI RegistrationCacheRepository

File: app/uk/gov/hmrc/pillar2/service/RegistrationService.scala [hunk 11]
```diff
@@ -70,19 +68,32 @@ class RegistrationService @Inject() (
...
+    hc:                                       HeaderCarrier
   ): Future[HttpResponse] = {
     val registerWithoutIDRequest = RegisterWithoutIDRequest(businessName, address, contactDetails)
     val response = dataSubmissionConnectors
       .sendWithoutIDInformation(registerWithoutIDRequest)(hc, ec)
-    response.map { res =>
-      val resReceived = AuditResponseReceived(res.status, res.json)
+    response.map { _ =>
       if (isFm) {
         val auditData = converNfmAuditDetails(registerWithoutIDRequest)
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:98]: correct typo in method name converNfmAuditDetails & converNfmAuditDetails

File: app/uk/gov/hmrc/pillar2/service/RegistrationService.scala [hunk 12]
```diff
@@ -52,12 +51,11 @@ class RegistrationService @Inject() (
       false
     )
   }.getOrElse {
-    logger.warn(s"[Session ID: ${SessionIdHelper.sessionId(hc)}] - RegistrationService - Upe Registration Information Missing")
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:55]: remove unused import for SessionIdHelper

File: app/uk/gov/hmrc/pillar2/models/identifiers/RfmNameRegistrationId.scala [hunk 13]
```diff
@@ -17,11 +17,10 @@
 package uk.gov.hmrc.pillar2.models.identifiers
 
 import play.api.libs.json.JsPath
-import uk.gov.hmrc.pillar2.models.RowStatus
 
-case object GrsUpeStatusId extends TypedIdentifier[RowStatus] {
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/pillar2/models/identifiers/RfmNameRegistrationId.scala:22]: Do we need to also rename occurrences of 'GrsUpStatus' in ModelGenerators

**Review by jassalrichy — APPROVED** [review]

**Review by javed-iqbal1 — APPROVED** [review]

**Pull request merged**